### PR TITLE
Feat/signup

### PR DESCRIPTION
**Pull request by MargaritaSukhova**
## Describe your changes

- Add Privacy Policy and Terms of Use pages and routes
- Add react-hook-form Controllers and validation to Sign Up page
- Add PolicyTermsLinks component and navigation to Sign Up page
- Updated error messages in Sign Up form
- Add service for user and userSlice in redux
- Updated styles for Alert in toast

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-6?atlOrigin=eyJpIjoiNDNmYTkyN2UxYWEwNGQzZGIwOTBkMzZkOGQ4ZTJiZTAiLCJwIjoiaiJ9)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.
Снимок экрана 2024-06-06 в 23 56 33
Снимок экрана 2024-06-06 в 23 57 02
Снимок экрана 2024-06-06 в 23 57 31
Снимок экрана 2024-06-06 в 23 58 18
Снимок экрана 2024-06-06 в 23 58 09
Снимок экрана 2024-06-06 в 23 57 46

Снимок экрана 2024-06-06 в 23 52 31
Снимок экрана 2024-06-06 в 23 52 42
Снимок экрана 2024-06-07 в 17 04 29

### CODE REVIEW

File: src/components/SignUpForm/index.tsx
Comment (lyudad) [inline src/components/SignUpForm/index.tsx:1]: I assume on signUp page you have only sign up form, why it in component folder then? I assume components should include reusable components only, no?

File: src/components/shared/PasswordInput/index.tsx [hunk 1]
```diff
@@ -1,47 +1,45 @@
...
-  const handleClickShowPassword = () => setShowPassword(!showPassword);
-  const handleMouseDownPassword = (
-    event: React.MouseEvent<HTMLButtonElement>
-  ) => {
-    event.preventDefault();
-  };
+const PasswordInput = forwardRef<HTMLInputElement, StyledInputProps>(
+  ({ placeholder, width, padding, name, stylevariant, ...props }, ref) => {
+    const [showPassword, setShowPassword] = useState(false);
+    const handleClickShowPassword = () => setShowPassword(!showPassword);
```
Comment (lyudad) [inline src/components/shared/PasswordInput/index.tsx]: empty line above, I assume team should add some rule to eslint to separate code, we can have everything without separation its difficult to read

File: src/components/SignUpForm/index.tsx
Comment (MargaritaSukhova) [reply src/components/SignUpForm/index.tsx:1]: Moved SignUpFrom to SignUpPage folder

File: src/components/shared/PasswordInput/index.tsx (hunk 1 above)
Comment (MargaritaSukhova) [reply src/components/shared/PasswordInput/index.tsx]: add new rule to eslint to have empty lines

File: .env.example [hunk 2]
```diff
@@ -1 +1 @@
-API_URL=
```
Comment (lyudad) [inline .env.example]: why it was renamed?

Comment (MargaritaSukhova) [reply .env.example]: Vite needs VITE prefix to env variables to download them automatically with no need to add some additional configs, and I decided to use this feature of Vite

Comment (lyudad) [reply .env.example]: got you

**Review by lyudad — APPROVED** [review]

File: src/pages/SignUpPage/SignUpForm/index.tsx
Comment (lyudad) [inline src/pages/SignUpPage/SignUpForm/index.tsx:1]: I assume its not your file, why its in pr?

Comment (MargaritaSukhova) [reply src/pages/SignUpPage/SignUpForm/index.tsx:1]: SignUp is my task in Jira and I worked over it all the time
I moved it so SignUp page folder

**Pull request merged**